File: solutions/0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) 
    {
        int l = 0, r = height.length - 1;

        int maxArea = 0;

        while (l < r)
        {
            int currentArea = (r - l) * Math.min(height[r], height[l]);

            int minHeight = Math.min(height[r], height[l]);

            if (minHeight == height[l])
            {
                l++;
            } else
            {
                r--;
            }

            maxArea = Math.max(maxArea, currentArea);
        }

        return maxArea;    
    }
}

File: solutions/0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        /*
            area = width (r - l) * height (height[i])
        */

        int max_area = 0;
        int l = 0, r = height.size() - 1;

        while (l < r) {
            int current_area = (r - l) * min(height[l], height[r]);
            max_area = max(max_area, current_area);

            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }

        return max_area;
    }
};
